new-tutorial/2_dglgraph.py

```diff
@@ -0,0 +1,211 @@
+#    Here are some general suggestions:
+# 
+#    -  For categorical attributes (e.g. gender, occupation), consider
+#       converting them to integers or one-hot encoding.
+#    -  For variable length string contents (e.g. news article, quote),
+#       consider applying a language model.
+#    -  For images, consider applying a vision model such as CNNs.
+# 
+#    You can find plenty of materials on how to encode such attributes
+#    into a tensor in the `PyTorch Deep Learning
+#    Tutorials <https://pytorch.org/tutorials/>`__.
+# 
+
+
+######################################################################
+# Graph Queries
+# -------------
+# 
+# ``DGLGraph`` object provides various methods to query a graph structure.
+# 
+
+print(g.num_nodes())
+print(g.num_edges())
+# Out degrees of the internal node
+print(g.out_degrees(0))
+# In degrees of the internal node - note that the graph is directed so the in degree should be 0.
+print(g.in_degrees(0))
+
+
+######################################################################
+# Graph Transformations
+# ---------------------
+# 
+
+
+######################################################################
+# You can also transform a graph to another graph.
+# 
+# A common transformation is to create a subgraph from the original graph:
+# 
+
+# Induce a subgraph from node 0, node 1 and node 3 from the original graph.
+sg1 = g.subgraph([0, 1, 3])
+# Induce a subgraph from edge 0, edge 1 and edge 3 from the original graph.
+sg2 = g.edge_subgraph([0, 1, 3])
+
+
+######################################################################
+# You can obtain the node/edge mapping from the subgraph to the original
+# graph by looking into the node feature ``dgl.NID`` or edge feature
+# ``dgl.EID`` in the new graph.
+# 
+
+# The original IDs of each node in sg1
+print(sg1.ndata[dgl.NID])
+# The original IDs of each edge in sg1
+print(sg1.edata[dgl.EID])
+# The original IDs of each node in sg2
+print(sg2.ndata[dgl.NID])
+# The original IDs of each edge in sg2
+print(sg2.edata[dgl.EID])
+
+
+######################################################################
+# Another common transformation is to add a reverse edge for each edge in
+# the original graph with ``dgl.add_reverse_edges``.
+# 
+# .. note::
+# 
+#    If you have an undirected graph, it is better to convert it
+#    into a bidirectional graph first via adding reverse edges.
+# 
+
+newg = dgl.add_reverse_edges(g)
+newg.edges()
+
+
+######################################################################
+# Loading and Saving Graphs
+# -------------------------
+# 
+# You can save a graph or a list of graphs via ``dgl.save_graphs`` and
+# load them back with ``dgl.load_graphs``.
+# 
+
+# Save graphs
+dgl.save_graphs('graph.dgl', g)
+dgl.save_graphs('graphs.dgl', [g, sg1, sg2])
+
+# Load graphs
+(g,), _ = dgl.load_graphs('graph.dgl')
+print(g)
+(g, sg1, sg2), _ = dgl.load_graphs('graphs.dgl')
+print(g)
+print(sg1)
+print(sg2)
```

Comment:
What is `_`?

Reply:
`dgl.load_graphs` normally returns a list of graphs and a dict of labels.  If `save_graphs` did not store any label, `dgl.load_graphs` will return an empty dict instead.  Hence the `_`.